src/web/js/trove/chart-lib.js

```diff
@@ -269,22 +396,251 @@
   }
 
   function barChart(globalOptions, rawData) {
+    // Variables and constants 
+    const table = get(rawData, 'tab');
+    var horizontal = get(rawData, 'horizontal');
+    const data = new google.visualization.DataTable();
+    var colors_list = [];
+    var default_color = "";
+    var pointers_list = [];
+    var pointer_color = 'black';
+    var axisTop, axisBottom, ticks;
+    var default_interval_color = 'black'; 
+
+    // Sets up the color list [Each Bar Colored Individually]
+    cases(RUNTIME.ffi.isOption, 'Option', get(rawData, 'colors'), {
+          none: function () {},
+          some: function (colors) {
+            colors_list = convertListWith(convertColor, colors);
+          }
+    });
+    const colors_list_length = colors_list.length;
+
+    // Sets up the default color [Default Bar Color if not specified in color_list]
+    cases(RUNTIME.ffi.isOption, 'Option', get(rawData, 'color'), {
+          none: function () {},
+          some: function (color) {
+            default_color = convertColor(color);
+          }
+    });
+
+    // Sets up the pointers list [Coloring each group memeber/stack]
+    cases(RUNTIME.ffi.isOption, 'Option', get(rawData, 'pointers'), {
+          none: function () {},
+          some: function (pointers) {
+            pointers_list = convertListWith(convertPointer, pointers);
+          }
+    });
+
+    // Sets up the pointer color
+    cases(RUNTIME.ffi.isOption, 'Option', get(rawData, 'pointer-color'), {
+          none: function () {},
+          some: function (color) {
+            pointer_color = convertColor(color);
+          }
+    });
+
+    // Sets up the default interval color
+    cases(RUNTIME.ffi.isOption, 'Option', get(rawData, 'default-interval-color'), {
+          none: function () {},
+          some: function (color) {
+            default_interval_color = convertColor(color);
+          }
+    });
+
+    // Sets up the calculated axis properties/data
+      cases(RUNTIME.ffi.isOption, 'Option', get(rawData, 'axisdata'), {
+          none: function () {},
+          some: function (axisdata) {
+            axisTop = toFixnum(axisdata.dict.axisTop);
+            axisBottom = toFixnum(axisdata.dict.axisBottom);
+            ticks = convertListWith(convertPointer, axisdata.dict.ticks);
+          }
+      });
+
+    // Initializes the Columns of the data 
+    data.addColumn('string', 'Label');
+    data.addColumn('number', 'Values');
+    data.addColumn({type: 'string', role: 'style'});
+
+    // Adds each row of bar data and bar_color data
+    table.forEach(function (row, idx) {
+      const bar_color = idx < colors_list_length ? colors_list[idx] : default_color;
+      data.addRow([row[0], toFixnum(row[1]), bar_color]);
+    });
+    addAnnotations(data, rawData);
+    addIntervals(data, rawData);
+
+    var options = {
```

Comment:
Use `let` or `const` if possible. Also, there's a missing semicolon.

Reply:
Got it! Will fix it in the next commit!